### ISSUE TITLE
Issue with EncryptCookies middleware, session not forgotten

### ISSUE THREAD

**Issue by mrk-j** [themsaid/laravel-mail-preview#45]
Steps to reproduce:

1. `laravel new mail-preview-test`
2. `composer require themsaid/laravel-mail-preview`
3. Create a new `MailPreviewController` with the following contents:

```php
<?php

namespace App\Http\Controllers;

use Mail;

class MailPreviewController extends Controller
{
    public function send()
    {
        Mail::raw('foo', function ($message) {
            $message->to('foo@bar.com');
        });

        var_dump(session()->all());

        return 'mail sent';
    }
    public function preview()
    {
        var_dump(session()->all());

        return '</body>';
    }
}
```

4. Change `web.php` to:

```php
<?php

Route::get('/', 'MailPreviewController@send');
Route::get('/preview', 'MailPreviewController@preview');
```

5. Visit http://mail-preview-test.test/
6. Visit http://mail-preview-test.test/preview
7. See that middleware to unset session is not working correctly, the preview link is visible after refreshing

Demo repository: https://github.com/mrk-j/laravel-mail-preview-bug-example

**Comment by mrk-j** [comment]
This seems related with this change Laravel 5.8:

```
Sessions
The StartSession Middleware
Likelihood Of Impact: Very Low

The session persistence logic has been moved from the terminate() method to the  handle() method. If you are overriding one or both of these methods, you should update them to reflect these changes.
```

The session is persisted before the this package's middleware has ran.

**Comment by mrk-j** [comment]
The PR in the `laravel/framework` repository: https://github.com/laravel/framework/pull/26410

**Comment by mrk-j** [comment]
This problem still exists with the latest release. Note that the problem only occurs when using the `EncryptCookies` middleware.

Can you maybe have a look @themsaid? I can update the PR #46 for version 2.1.

**Comment by mrk-j** [comment]
Fixed after merging #46.
